### PR TITLE
Fixed bug when logged in customer can access login/registration p…

### PR DESCRIPTION
**Pull request by sarjon**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Fixes bug when logged in customer can access login/registration pages.
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | n/a
| How to test?  | Log in to Front Office and try accessing login or registration pages.

### CODE REVIEW

**Comment by maximebiloe** [comment]
Hello @sarjon,

What is supposed to do after applying your fix?

Regards

**Comment by sarjon** [comment]
Well, if customer is logged in, he shouldn't be able to access login/registration pages, since it doesn't make any sense. This fix redirects customer to account page instead.

File: controllers/front/AuthController.php [hunk 1]
```diff
@@ -30,6 +30,16 @@ class AuthControllerCore extends FrontController
     public $php_self = 'authentication';
     public $auth = false;
 
+    public function checkAccess()
+    {
+        if ($this->context->customer->isLogged() && !$this->ajax) {
+            $this->redirect_after = ($this->authRedirection) ? urlencode($this->authRedirection) : 'my-account';
```
Comment (kpodemski) [inline controllers/front/AuthController.php]: i'd use $link->getPageLink as rewrite can be different

Comment (sarjon) [inline controllers/front/AuthController.php]: It uses $link->getPageLink behide the scene.

Comment (antoin-m) [inline controllers/front/AuthController.php]: Yes, `Tools::redirectLink` actually uses `getPageLink` under the hood :)

**Comment by antoin-m** [comment]
Thank you @sarjon :)

**Pull request merged**